sdk/python/feast/offline_store.py

```diff
@@ -11,18 +11,76 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import time
 from abc import ABC, abstractmethod
-from datetime import datetime
-from typing import List, Optional, Tuple
+from dataclasses import asdict, dataclass
+from datetime import datetime, timedelta
+from typing import Dict, List, Optional, Tuple, Union
 
+import pandas as pd
 import pyarrow
+from google.cloud import bigquery
+from jinja2 import BaseLoader, Environment
 
+from feast.data_source import BigQuerySource, FileSource
 from feast.feature_view import FeatureView
+from feast.repo_config import RepoConfig
+
+ENTITY_DF_EVENT_TIMESTAMP_COL = "event_timestamp"
+
+
+class RetrievalJob(ABC):
+    """RetrievalJob is used to manage the execution of a historical feature retrieval"""
+
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def to_df(self):
+        """Return dataset as Pandas DataFrame synchronously"""
+        pass
+
+
+class FileRetrievalJob(RetrievalJob):
+    def __init__(self, evaluation_function):
```

Comment:
I'd add a type signature for `evaluation_function` here, and a comment explaining what it is for.